git-server/Cargo.toml

```diff
@@ -6,16 +6,31 @@ authors = ["Alexis Sellier <self@cloudhead.io>"]
 edition = "2018"
 build = "../build.rs"
 
+[[bin]]
+name = "pre-receive"
+path = "bin/pre_receive.rs"
+
+[[bin]]
+name = "post-receive"
+path = "bin/post_receive.rs"
+
 [dependencies]
 base64 = { version = "0.13" }
 shared = { path = "../shared", default-features = false }
 warp = { version = "0.3.1", features = ["tls"] }
 http = { version = "0.2" }
-thiserror = { version = "1" }
+thiserror = { version = "1.0.30" }
 tokio = { version = "1.2", features = ["macros", "rt", "rt-multi-thread", "sync"] }
 argh = { version = "0.1.4" }
 tracing = "0.1"
 tracing-subscriber = "0.2"
+git2 = "0.13.20"
+envconfig = "0.10.0"
+pgp = "0.7.2"
+hex = "0.4.3"
+simplelog = "0.10.2"
+log = "0.4.14"
```

Comment:
For small binaries that don't need real logging, I prefer to use `eprintln`/`println`

Reply:
> For small binaries that don't need real logging, I prefer to use `eprintln`/`println`

One of the nice things about using the simplelog terminal logger is having the colored log levels, so we can notify the sender of any conditions they might need to satisfy the authentication, e.g. signing pushes, adding keys to the keyring or re-submitting a signed push because of a SLOP nonce certificate. 

Here's what it looks like when an invalid nonce is submitted. Some thing I would probably change here is moving the `info` levels to `debug`; I just didn't know how to tell the git-server to set the `RUST_LOG=debug` flag when invoking the hook.

```
Enumerating objects: 67, done.
Counting objects: 100% (67/67), done.
Delta compression using up to 16 threads
Compressing objects: 100% (51/51), done.
Writing objects: 100% (66/66), 22.91 KiB | 3.82 MiB/s, done.
Total 66 (delta 3), reused 0 (delta 0), pack-reused 0
remote: 15:39:57 [INFO] pre-receive: hook invoked
remote: 15:39:57 [ERROR] pre-receive: received invalid certificate nonce status: "SLOP"
remote: 15:39:57 [WARN] pre-receive: please re-submit signed push to request certificate
remote: 15:39:57 [ERROR] pre-receive: FailedCertificateVerification
To http://0.0.0.0:8778/hnrkbtw9t1of4ykjy6er4qqwxtc54k9943eto
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'http://0.0.0.0:8778/hnrkbtw9t1of4ykjy6er4qqwxtc54k9943eto'
```

![Screen Shot 2021-10-18 at 8 42 22 AM](https://user-images.githubusercontent.com/2837196/137764492-6e240043-4c82-4eae-a559-6099cc6c0ee1.png)


That said, the file logs can probably be removed. I don't think there is any reason to keep these logs, other than keeping a log for debugging errors, if needed. But I think file logging should be an optional flag.